Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -22,239 +24,246 @@ class Client(BaseClient):
-            last_time_date = arg_to_datetime(arg=last_time, required=True).date()   # type: ignore[union-attr]
-            event_date = arg_to_datetime(arg=inc_time, required=True).date() if inc_time else None   # type: ignore[union-attr]
-            if (inc_time and inc_time < last_time) or event.get("dg_guid") in id_list:
-                continue
-            if last_time_date == event_date and event.get("dg_guid"):
-                id_list.append(event.get("dg_guid"))
-            elif event.get("dg_guid"):
-                id_list = [event.get("dg_guid")]
-        else:
-            if event.get("dg_guid"):
-                id_list.append(event.get("dg_guid"))
-        event_list_for_push.append(event)
-        last_time = inc_time if inc_time else last_time
-        index += 1
-        if index == limit:
+
+    for event_data in events_data:
+        if limit is not None and index == limit:  # No API-side limit param (needs to be managed on our end)
             break
-    return event_list_for_push, last_time, id_list
+
+        event = dict(zip(event_fields, event_data, strict=True))
+        event_id = event['dg_guid']
+
+        if event_id in ids:
+            continue  # Skip duplicate event
+
+        event_time = arg_to_datetime(arg=event.get('dg_time')) if event.get('dg_time') else None
+        event['_time'] = event_time.strftime(DATE_FORMAT) if event_time else None
+
+        ids.add(event_id)
+        index += 1
+
+        yield event
 
 
-def fetch_events(client: Client, last_run: dict[str, list], limit: int) -> Tuple[dict, list]:
+def get_events_command(client: Client, args: dict) -> tuple[list, dict, CommandResults]:
     """
+    Fetches a limited number of events and returns it in the CommandResults as a human readable markdown table.
     Args:
-        client (Client): DG client to use.
-        last_run (dict): A dict with a key containing the latest event created time we got from last fetch.
-        limit (int):
-
+        client (Client): Digital Guardian client.
+        args (dict): Command arguments.
     Returns:
-        dict: Next run dictionary containing the timestamp that will be used in ``last_run`` on the next fetch and list of ID's
-              of ingested id's
-        list: List of events that will be created in XSIAM.
+        tuple[list, dict, CommandResults]: List of events, last run dictionary, CommandResults with human readable output.
     """
-    id_list = last_run.get('id_list', [])
-    last_time = str(last_run.get('start_time') or "")
-    demisto.debug('fetching events')
-    event_list = get_raw_events(client, last_time)
-    event_list_for_push, time_of_event, id_list = create_events_for_push(event_list, last_time, id_list, limit)
+    limit = arg_to_number(args.get('limit')) or 1000
+    events, last_run = fetch_events(client, limit=limit)
+
+    event_list = list(events)
+    if event_list:
+        human_readable = tableToMarkdown(
+            name='Test Events',
```

Comment:
`Test`?

Reply:
Now showing `"Events for Profile {export_profile}"` in commit 582763c169821e56b0a063bc42e00239eb151efc